src/foam/nanos/alarming/AlarmLoggingDAO.js

```diff
@@ -12,14 +12,22 @@ foam.CLASS({
   documentation: `Generate Logger messages for each alarm`,
 
   javaImports: [
+    'foam.dao.DAO',
     'foam.nanos.logger.Logger',
+    'foam.nanos.notification.Notification'
   ],
 
   methods: [
     {
       name: 'put_',
       javaCode: `
+      Alarm old = (Alarm) getDelegate().find_(x, obj.getProperty("id"));
```

Comment:
Is this code generic, or you always know the type of the object? If you know, you could just cast it and then do .getID(), which would be slightly faster.

Reply:
I mean, you could just cast 'obj' to Alarm.